index.js

```diff
@@ -16,15 +21,56 @@ async function run() {
     const logConfigurationOptions = core.getInput("log-configuration-options", { required: false });
     const dockerLabels = core.getInput('docker-labels', { required: false });
     const command = core.getInput('command', { required: false });
+    
+     //New inputs to fecth task definition 
+     const taskDefinitionArn = core.getInput('task-definition-arn', { required: false });
+     const taskDefinitionFamily = core.getInput('task-definition-family', { required: false });
+     const taskDefinitionRevision = Number(core.getInput('task-definition-revision', { required: false}));
 
-    // Parse the task definition
-    const taskDefPath = path.isAbsolute(taskDefinitionFile) ?
-      taskDefinitionFile :
-      path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
-    if (!fs.existsSync(taskDefPath)) {
-      throw new Error(`Task definition file does not exist: ${taskDefinitionFile}`);
+
+    if(taskDefinitionFile && taskDefinitionArn && taskDefinitionFamily){ 
+      core.warning("Task definition file will be option used.");
+    }
+    if(taskDefinitionFamily && !taskDefinitionRevision){
+      core.warning("The latest revision of the task definition family will be provided");
     }
-    const taskDefContents = require(taskDefPath);
+    if(taskDefinitionRevision){
+      core.setFailed("You cant fetch task definition with just revision: Either use task definition file, arn or family");
+    }
+  
+    let taskDefPath;
+    let describeTaskDefResponse;
+
+    if(taskDefinitionFile){
+    taskDefPath = path.isAbsolute(taskDefinitionFile) ?
+      taskDefinitionFile : 
+      path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile); 
+
+      if (!fs.existsSync(taskDefPath)) {
+        throw new Error(`Task definition file does not exist: ${taskDefinitionFile}`);
+      }
+    }
+
+    else if((taskDefinitionArn) || (taskDefinitionFamily && taskDefinitionRevision) || (taskDefinitionFamily && !taskDefinitionRevision)){
+      try{
+        describeTaskDefResponse = await ecs.describeTaskDefinition({
+        taskDefArn: taskDefinitionArn,
+        taskDefFam: taskDefinitionFamily,
```

Comment:
What would the value of "taskDefinitionFamily" be is it not provided by the user. Would it be a empty string or a null value? 

Should be checking any of the inputs to describeTaskDefinition API (taskDefinitionFamily, taskDefinitionRevision, or taskDefinitionArn) are null/empty before calling the `ecs.describeTaskDefinition`. Or would this null/empty check happen anywhere else.